prover/src/lib.rs

```diff
@@ -326,46 +327,28 @@ pub trait Prover {
         // 2 ----- evaluate constraints -----------------------------------------------------------
         // evaluate constraints specified by the AIR over the constraint evaluation domain, and
         // compute random linear combinations of these evaluations using coefficients drawn from
-        // the channel; this step evaluates only constraint numerators, thus, only constraints with
-        // identical denominators are merged together. the results are saved into a constraint
-        // evaluation table where each column contains merged evaluations of constraints with
-        // identical denominators.
+        // the channel
         #[cfg(feature = "std")]
         let now = Instant::now();
         let constraint_coeffs = channel.get_constraint_composition_coeffs();
         let evaluator = self.new_evaluator(&air, aux_trace_rand_elements, constraint_coeffs);
-        let constraint_evaluations = evaluator.evaluate(&trace_lde, &domain);
+        let composition_poly_trace = evaluator.evaluate(&trace_lde, &domain);
         #[cfg(feature = "std")]
         debug!(
             "Evaluated constraints over domain of 2^{} elements in {} ms",
-            constraint_evaluations.num_rows().ilog2(),
+            composition_poly_trace.num_rows().ilog2(),
             now.elapsed().as_millis()
         );
 
         // 3 ----- commit to constraint evaluations -----------------------------------------------
 
-        // first, build constraint composition polynomial from the constraint evaluation table:
-        // - divide all constraint evaluation columns by their respective divisors
-        // - combine them into a single column of evaluations,
-        // - interpolate the column into a polynomial in coefficient form
-        // - "break" the polynomial into a set of column polynomials each of degree equal to
-        //   trace_length - 1
-        #[cfg(feature = "std")]
-        let now = Instant::now();
-        let composition_poly =
-            constraint_evaluations.into_poly(air.context().num_constraint_composition_columns())?;
-        #[cfg(feature = "std")]
-        debug!(
-            "Converted constraint evaluations into {} composition polynomial columns of degree {} in {} ms",
-            composition_poly.num_columns(),
-            composition_poly.column_degree(),
-            now.elapsed().as_millis()
+        // first, build a commitment to the evaluations of the composition polynomial columns
```

Comment:
nit: maybe add constraint to "composition polynomial columns" in order to retain some consistency?

Reply:
agreed! added!